vault/core.go

```diff
@@ -1239,9 +1244,9 @@ func (c *Core) Shutdown() error {
 
 	c.stateLock.Lock()
 	defer c.stateLock.Unlock()
-	if c.shutdownDoneCh != nil {
-		close(c.shutdownDoneCh)
-		c.shutdownDoneCh = nil
+	if c.shutdownDoneCh.Load().(chan struct{}) != nil {
```

Comment:
This is racy: you shouldn't access an atomic in an if condition and then again in the body like this.  Instead, assign the Load() to a local var, then look at that var in both the condition and the body.